scripts/imgtool/keys/general.py

```diff
@@ -9,17 +9,19 @@
 
 class KeyClass(object):
     def _emit(self, header, trailer, encoded_bytes, indent, file=sys.stdout, len_format=None):
-        print(AUTOGEN_MESSAGE, file=file)
-        print(header, end='', file=file)
-        for count, b in enumerate(encoded_bytes):
-            if count % 8 == 0:
-                print("\n" + indent, end='', file=file)
-            else:
-                print(" ", end='', file=file)
-            print("0x{:02x},".format(b), end='', file=file)
-        print("\n" + trailer, file=file)
-        if len_format is not None:
-            print(len_format.format(len(encoded_bytes)), file=file)
+        sys.stdout.close = lambda: None
```

Comment:
The `sys.stdout` does not have the close method implemented and would throw an error when the `with (open...`'s scope ends. This way we add an empty method to prevent this error and there's no need for additional checks whether the `file` is `sys.stdout` or a regular file.

Reply:
I've tried it without the `sys.stdout.close = lambda: None` line and it worked fine. Isn't it that with the `is not sys.stdout` check it's avoided?

Reply:
You are right, my reasoning was not correct. :grimacing: 
`sys.stdout` does have a `close()` method but once it's called, the `print()` will no longer work until the `sys.stdout`'s `open()` is called again. 
So a print() would cause a crash if this `sys.stdout.close = lambda: None` is removed so as a workaround its `close()` is overridden.

Reply:
Removed this solution from the latest patchset because it'd change the global state and it's not that clear either.

Reply:
Thanks for the update.
Another solution could have been to use the open()/close() methods directly, without `with`.